### ISSUE TITLE
Conversion to 3D: A clear semantics.

### ISSUE THREAD

**Issue by backspaces** [backspaces/asx#24]
We need a coherent strategy for conversion from planar x,y coordinates to x,y,z.

Ex: The turtle setxy(x,y) method. Currently we set a Default value for z by converting setxy to having an optional z value. If not given, the default z value is used. If z given, the turtle promotes the z default to an instance value for z.

Distance computations currently are all x,y based. They need to use z if available.

The "inRect" and "inRadius" methods: should we have a new set more appropriately named "inCube" and "inSphere".

Patches, which act as a quadTree also come into play. Should a turtle's z value be relative to the patch z/elevation value? 

Patches currently are a square World.patchSize (likely to be 1 in the future: #20) on a side centered at the center of the patch. The corners of the square are presumed to be a bi-linear interpolation of the adjoining squares. This allows "terraining" of the patches when draped over a Three segmented Plane geometry.

If a turtle asks for it's patch coordinate, should it be the patch center? Or the bilinear interpolation based on the turtle's float x,y? (I'd say centered, but interpolation when using DataSet convolutions/sampling)

In other words, we need a clearer strategy for our 3D conversion.

**Comment by Vandivier** [comment]
Speaking as a JS addict but also someone who's not a particular fan of NetLogo, I would prefer plain English method names to NetLogo-like naming. Completely understood if this view is not compatible with the lib design.

So when describing the position of a patch or turtle I would prefer something like instance.position(x, y, z) using a class setter or AS.position(instance, x, y, z) using the module object setter. Getters like instance.position().

To be clear, inRadius !== inSphere. inRadius could be used for a cone or a cylinder, etc. inSphere is a great addition, but not a replacement.

At a strategic level, per the above, I would implement 3D transforms as an optional module since a model can be implemented without that.
